### PR TITLE
docs(free-programming-books-subjects): fixed broken links #5470

### PR DESCRIPTION
**Pull request by LuigiImVector**
## What does this PR do?
Add info | Improve repo

### Description

I fixed the broken links on `free-programming-books-subjects.md`, see #5470.

False-positive links: 07, 11, 12, 15

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -441,7 +441,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Non-Uniform Random Variate Generation](http://luc.devroye.org/rnbookindex.html) - Luc Devroye (PDF)
 * [Notes on Diffy Qs](https://www.jirka.org/diffyqs/) - Jiří Lebl
 * [Number Theory](https://github.com/holdenlee/number-theory) - Holden Lee MIT
-* [Number Theory: In Context and Interactive](http://math.gordon.edu/ntic/) - Karl-Dieter Crisman
+* [Number Theory: In Context and Interactive](https://web.archive.org/web/20210711231452/https://math.gordon.edu/ntic/) - Karl-Dieter Crisman
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:444]: Added

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -240,7 +240,6 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Introduction to Data Science](https://docs.google.com/file/d/0B6iefdnF22XQeVZDSkxjZ0Z5VUE/edit?pli=1) - Jeffrey Stanton
 * [Introduction to Data Science](https://leanpub.com/datasciencebook) - Rafael A Irizarry *(Leanpub account or valid email requested)*
 * [Mining of Massive Datasets](http://infolab.stanford.edu/~ullman/mmds/book.pdf) - Jure Leskovec, Anand Rajaraman, Jeffrey D. Ullman (PDF)
-* [School of Data Handbook](http://schoolofdata.org/handbook/)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:243]: My vote: yes

File: books/free-programming-books-subjects.md [hunk 3]
```diff
@@ -711,8 +709,8 @@ Kerridge (PDF) (email address *requested*, not required)
 * [How to Write Unmaintainable Code](http://mindprod.com/jgloss/unmain.html)
 * [Kanban and Scrum - making the most of both](http://www.infoq.com/minibooks/kanban-scrum-minibook)
 * [Kanban for skeptics](https://leanpub.com/kanbanforskeptics) - Nick Oostvogels  *(Leanpub account or valid email requested)*
-* [Microservices AntiPatterns and Pitfalls](https://www.oreilly.com/programming/free/files/microservices-antipatterns-and-pitfalls.pdf) (PDF)
-* [Microservices vs. Service-Oriented Architecture](https://www.oreilly.com/programming/free/files/microservices-vs-service-oriented-architecture.pdf) (PDF)
+* [Microservices AntiPatterns and Pitfalls](https://www.oreilly.com/content/microservices-antipatterns-and-pitfalls/) - Mark Richards (HTML)
```
[the remark this answers is not in the document]
Comment (eshellman) [reply books/free-programming-books-subjects.md:712]: just the PDF

File: books/free-programming-books-subjects.md [hunk 4]
```diff
@@ -711,8 +709,8 @@ Kerridge (PDF) (email address *requested*, not required)
...
+* [Microservices vs. Service-Oriented Architecture](https://web.archive.org/web/20201112010849/https://www.oreilly.com/programming/free/files/microservices-vs-service-oriented-architecture.pdf) - Mark Richards (PDF) *(:card_file_box: archived)*
```
[the remark this answers is not in the document]
Comment (eshellman) [reply books/free-programming-books-subjects.md:713]: leave only main link

File: books/free-programming-books-subjects.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (eshellman) [reply books/free-programming-books-subjects.md:444]: yes, remove for now

File: books/free-programming-books-subjects.md (hunk 2 above)
Comment (davorpa) [inline books/free-programming-books-subjects.md:243]: Recover `School of Data Handbook`

```suggestion
* [Mining of Massive Datasets](http://infolab.stanford.edu/~ullman/mmds/book.pdf) - Jure Leskovec, Anand Rajaraman, Jeffrey D. Ullman (PDF)
* [School of Data Handbook](http://schoolofdata.org/handbook/)
```

File: books/free-programming-books-subjects.md [hunk 5]
```diff
@@ -240,7 +240,6 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 * [Introduction to Data Science](https://docs.google.com/file/d/0B6iefdnF22XQeVZDSkxjZ0Z5VUE/edit?pli=1) - Jeffrey Stanton
 * [Introduction to Data Science](https://leanpub.com/datasciencebook) - Rafael A Irizarry *(Leanpub account or valid email requested)*
 * [Mining of Massive Datasets](http://infolab.stanford.edu/~ullman/mmds/book.pdf) - Jure Leskovec, Anand Rajaraman, Jeffrey D. Ullman (PDF)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:242]: Recover `School of Data Handbook`

```suggestion
* [Mining of Massive Datasets](http://infolab.stanford.edu/~ullman/mmds/book.pdf) - Jure Leskovec, Anand Rajaraman, Jeffrey D. Ullman (PDF)
* [School of Data Handbook](http://schoolofdata.org/handbook/)
```

File: books/free-programming-books-subjects.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md]: Resolved at https://github.com/EbookFoundation/free-programming-books/pull/6471/commits/18b7bb34e7740422a6342a935c9398213e377ad3

File: books/free-programming-books-subjects.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:713]: Suggestion changed to PDF.

Frontend: https://www.oreilly.com/content/microservices-antipatterns-and-pitfalls/
Archived PDF: http://web.archive.org/web/20210205164251/https://www.oreilly.com/programming/free/files/microservices-antipatterns-and-pitfalls.pdf

File: books/free-programming-books-subjects.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md]: Suggestion updated and applied in https://github.com/EbookFoundation/free-programming-books/pull/6471/commits/5b91d26b9300f48d52b91cef24266613c87b204c

File: books/free-programming-books-subjects.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:445]: > yes, remove for now

Archived link works fine for me. Frontmatter and PDF download continues available

Adding formats and notes
```suggestion
* [Number Theory: In Context and Interactive](https://web.archive.org/web/20210711231452/https://math.gordon.edu/ntic/) - Karl-Dieter Crisman (HTML, PDF) *(:card_file_box: archived)* 
```

[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:445]: Or remove...
```suggestion
```

[the remark this answers is not in the document]
Comment (davorpa) [reply books/free-programming-books-subjects.md:445]: Available again https://math.gordon.edu/ntic/ :wink:

**Pull request merged**